torch/csrc/utils/tensor_new.cpp

```diff
@@ -141,6 +141,9 @@ ScalarType infer_scalar_type(PyObject *obj) {
   if (PyArray_Check(obj)) {
     auto array = (PyArrayObject*)obj;
     return numpy_dtype_to_aten(PyArray_TYPE(array));
+  } else if (PyArray_CheckScalar(obj)) {
```

Comment:
nit: use an if rather than an else if here to match the code around it

Reply:
Sure, thank you for pointing it out.